### PR TITLE
[RLlib] Add `ObservationPreprocessor` (ConnectorV2) and cleanup ConnectorV2 docstrings.

### PR DESCRIPTION
**Pull request by sven1977**
Add `ObservationPreprocessor` (ConnectorV2) and cleanup ConnectorV2 docstrings.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/connectors/common/batch_individual_items.py [hunk 1]
```diff
@@ -10,9 +10,93 @@
...
+        }
+
+        # Create our (single-agent) connector piece.
+        connector = BatchIndividualItems()
+
+        # Call the connector (and thereby batch the individual items).
+        output_data = connector(
+            rl_module=None,  # This particular connector works without an RLModule.
+            data=single_agent_data,
+            episodes=[],  # This particular connector works without a list of episodes.
```
Comment (simonsays1980) [inline rllib/connectors/common/batch_individual_items.py:87]: Can we put comments always in between code and not after it? Imo this makes it better readable and easier for formatting. What do you think?

File: rllib/connectors/env_to_module/observation_preprocessor.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+from ray.rllib.connectors.connector_v2 import ConnectorV2
+from ray.rllib.core.rl_module.rl_module import RLModule
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.typing import EpisodeType
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class ObservationPreprocessor(ConnectorV2, abc.ABC):
```
Comment (simonsays1980) [inline rllib/connectors/env_to_module/observation_preprocessor.py:12]: Awesome!

**Review by simonsays1980 — APPROVED** [review]
LGTM.

File: rllib/connectors/common/batch_individual_items.py (hunk 1 above)
Comment (sven1977) [reply rllib/connectors/common/batch_individual_items.py:87]: Good point, especially here, where it's a long comment! Will change ...

Sometimes, I feel it's ok to put a few-word comment right after the line, if it fits.

**Pull request merged**